pkg/tfvars/internal/cache/cache.go

```diff
@@ -119,16 +122,25 @@ func cacheFile(reader io.Reader, filePath string, sha256Checksum string) (err er
 	}
 
 	reader = io.MultiReader(bytes.NewReader(buf), reader)
-	fileType := http.DetectContentType(buf)
-	logrus.Debugf("content type of %s is %s", filePath, fileType)
-	switch fileType {
+	fileType, err := filetype.Match(buf)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("content type of %s is %s", filePath, fileType.MIME.Value)
+	switch fileType.MIME.Value {
```

Comment:
As an alternative to guessing the MIMEtype literal, we could get rid of the consts (and the generic `filetype.Match`) and use the matchers of our new dependency:
```
	switch {
	case matchers.Gz(buf):
		logrus.Debug("decompressing the image archive as gz")
		uncompressor, err := gzip.NewReader(reader)
		if err != nil {
			return err
		}
		defer uncompressor.Close()
		reader = uncompressor
	case matchers.Xz(buf):
		logrus.Debug("decompressing the image archive as xz")
		uncompressor, err := xz.NewReader(reader)
		if err != nil {
			return err
		}
		reader = uncompressor
	default:
		logrus.Debug("detected uncompressed image")
		// No need for an interposer otherwise
	}
```

Reply:
done :)